src/fromager/sources.py

```diff
@@ -122,18 +124,40 @@ def default_download_source(
     sdist_server_url: str,
 ) -> tuple[pathlib.Path, str, str]:
     "Download the requirement and return the name of the output path."
-    url, version = resolve_dist(
-        ctx, req, sdist_server_url, include_sdists=True, include_wheels=False
+    custom_download_urls = ctx.settings.url_to_download_sdist_from(req.name)
+    rename_to_template = ctx.settings.rename_sdist_to(req.name)
+
+    # don't include sdists if the user wants to download source from a predefined url
```

Comment:
If the user is providing a url from which they want to download the sdist then the url that the resolver returns for the source dist won't be used anyways

Reply:
Including or excluding specific formats influences the resolver, too. It will only return a version the distribution type we've asked for exists for that version.

I guess in most cases where we download source from an alternative location there isn't an sdist to use, so this is probably OK. We may end up wanting to expose options to control which types of distributions to consider, though. Let's deal with that in a separate PR. In fact, we may want to split the step for resolving the version from the step for downloading the source for that version.

It would be really good to see a functioning version of this PR with a real use case based on our downstream builds.